lib/components/NavigationBar/NavigationBar.test.tsx

```diff
@@ -0,0 +1,120 @@
+import { afterEach, describe, it, expect } from 'vitest';
+import { render, screen, cleanup } from '@testing-library/react';
+import '@testing-library/jest-dom/vitest';
+
+import { NavigationBar, NavigationBarLinkProps } from './NavigationBar';
+
+afterEach(() => {
+  cleanup();
+});
+
+describe('NavigationBar', () => {
+  const items = [
+    {
+      text: 'Home',
+      active: true,
+      href: '/home',
+    },
+    {
+      text: 'About',
+      active: false,
+      href: '/about',
+    },
+    {
+      text: 'Contact',
+      active: false,
+      href: '/contact',
+    },
+  ].map(({ text, active, href }) => ({
+    text,
+    active,
+    href,
+    component: ({ children, ...rootProps }: NavigationBarLinkProps) => (
+      <a href={href} {...rootProps}>
+        {children}
+      </a>
+    ),
+  }));
+
+  const user = {
+    name: 'Jane Doe',
+    src: 'user.jpg',
+    component: ({ children, ...rootProps }: NavigationBarLinkProps) => (
+      <a href="/profile" {...rootProps}>
+        {children}
+      </a>
+    ),
+  };
+
+  it('renders navigation items and user', () => {
+    const { container } = render(<NavigationBar items={items} user={user} />);
+
+    // Assert snapshot
```

Comment:
These `// Assert` comments might not bring enough value in most of the places, but at least they are short so they are most likely to be up to date in the future also.

Reply:
Agree. At least the blank lines between the code groups make it a little easier to read the code. I have no opinion on this.